flink-python/pyflink/proto/flink-fn-execution.proto

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package org.apache.flink.fn_execution.v1;
+
+option java_package = "org.apache.flink.fnexecution.v1";
+option java_outer_classname = "FlinkFnApi";
+
+// User-defined function definition. It supports chaining functions, that's, the execution
+// result of one user-defined function as the input of another user-defined function.
+message UserDefinedFunction {
+    message Input {
+        oneof input {
+            UserDefinedFunction udf = 1;
+            int32 inputOffset = 2;
+        }
+    }
+
+    // The serialized representation of the user-defined function
+    bytes payload = 1;
+
+    // The input arguments of the user-defined function, it could be one of the following:
+    // 1. A column from the input row
+    // 2. The result of another user-defined function
+    repeated Input inputs = 2;
+}
+
+// A list of user-defined functions to be executed in a batch.
+message UserDefinedFunctions {
+    repeated UserDefinedFunction udfs = 1;
+}
+
+// A representation of the data schema.
+message Schema {
+    enum TypeName {
+        TINYINT = 0;
+        SMALLINT = 1;
+        INT = 2;
+        BIGINT = 3;
+        DECIMAL = 4;
+        FLOAT = 5;
+        DOUBLE = 6;
+        DATE = 7;
+        TIME = 8;
+        DATETIME = 9;
+        BOOLEAN = 10;
+        BINARY = 11;
+        VARBINARY = 12;
+        CHAR = 13;
+        VARCHAR = 14;
+        ARRAY = 15;
+        MAP = 16;
+        MULTISET = 17;
+        ROW = 18;
```

Comment:
Row is a special type and it could be used to represent the type of a table. So it seems more reasonable to put it at last of the type list. What do you think?

Reply:
New types would be added later, and we can't change the tag number for ROW at that time. In this case, we can't make sure it is at the last. 
If we want to separate ROW from the others, we can put ROW at the front, this can also make encode ROW one less byte.(My main concern is the performance, see [here](https://developers.google.com/protocol-buffers/docs/pythontutorial) about the tag number)
What do you think?

Reply:
Good catch for the performance issue. Make sense to me. +1 to put it at the first of the list.